File: backend/internal/model/wizard.go
package model

// Questionnaire holds the user's answers from the site building wizard.
type Questionnaire struct {
	// Industry is the user's industry/business type (e.g., "technology", "consulting", "retail").
	Industry string `json:"industry" binding:"required"`

	// StylePreference is the visual style preference (e.g., "modern", "classic", "minimalist", "bold").
	StylePreference string `json:"stylePreference"`

	// Features is a list of desired functionality (e.g., ["blog", "contact_form", "portfolio"]).
	Features []string `json:"features"`

	// ContentTypes describes what kind of content the site will feature
	// (e.g., ["articles", "case_studies", "team_profiles"]).
	ContentTypes []string `json:"contentTypes"`

	// BrandName is the name of the brand or company.
	BrandName string `json:"brandName"`

	// Description is a short description of the business or site purpose.
	Description string `json:"description"`

	// Locale is the primary locale for generated content ("zh" or "en"). Defaults to "zh".
	Locale string `json:"locale"`
}

// PagePlan describes a single page to be scaffolded.
type PagePlan struct {
	// Slug is the URL slug for the page.
	Slug string `json:"slug"`

	// Title holds bilingual titles (keyed by locale, e.g., "zh"/"en").
	Title map[string]string `json:"title"`

	// Layout is the recommended layout type (e.g., "hero-cta", "grid", "split").
	Layout string `json:"layout"`

	// Sections lists the recommended section types (e.g., ["hero", "features", "contact"]).
	Sections []string `json:"sections"`

	// SortOrder is the display order of the page in navigation.
	SortOrder int `json:"sortOrder"`
}

// ColorScheme holds a recommended brand color palette.
type ColorScheme struct {
	// Primary is the main brand color in hex (e.g., "#2563EB").
	Primary string `json:"primary"`

	// Secondary is the accent color in hex.
	Secondary string `json:"secondary"`

	// Background is the recommended page background color.
	Background string `json:"background"`

	// Text is the recommended body text color.
	Text string `json:"text"`

	// Rationale explains why this palette was chosen.
	Rationale string `json:"rationale"`
}

// SuggestedContent holds AI-generated sample content for a specific page.
type SuggestedContent struct {
	// PageSlug is the target page slug this content is intended for.
	PageSlug string `json:"pageSlug"`

	// Heading is the main headline.
	Heading string `json:"heading"`

	// Subheading is the supporting subtitle or tagline.
	Subheading string `json:"subheading"`

	// Body is the main body copy.
	Body string `json:"body"`

	// CTAText is the suggested call-to-action button text.
	CTAText string `json:"ctaText"`
}

// SitePlan is the AI-generated site structure recommendation.
type SitePlan struct {
	// RecommendedTheme is the theme package name (e.g., "default", "modern-dark", "warm-earth").
	RecommendedTheme string `json:"recommendedTheme"`

	// Pages is the list of pages to scaffold.
	Pages []PagePlan `json:"pages"`

	// ColorScheme is the recommended color palette.
	ColorScheme ColorScheme `json:"colorScheme"`

	// SuggestedContent contains AI-generated sample content per page.
	SuggestedContent []SuggestedContent `json:"suggestedContent"`

	// Rationale explains the overall design and structure decisions.
	Rationale string `json:"rationale"`
}

// ScaffoldResult holds the outcome of applying a site plan.
type ScaffoldResult struct {
	// CreatedPages lists the slugs of pages successfully created.
	CreatedPages []string `json:"createdPages"`

	// SkippedPages lists slugs that were skipped because they already existed.
	SkippedPages []string `json:"skippedPages"`

	// AppliedTheme is the theme that was recommended (theme activation is handled separately).
	AppliedTheme string `json:"appliedTheme"`
}

// ColorSuggestionRequest is the input for the suggest-colors endpoint.
type ColorSuggestionRequest struct {
	Industry  string `json:"industry" binding:"required"`
	BrandName string `json:"brandName"`
	Locale    string `json:"locale"`
}

// GenerateContentRequest is the input for the generate-content endpoint.
type GenerateContentRequest struct {
	PageType    string `json:"pageType" binding:"required"`
	Industry    string `json:"industry" binding:"required"`
	BrandName   string `json:"brandName"`
	Description string `json:"description"`
	Locale      string `json:"locale"`
}

File: frontend/src/api/wizard.ts
import { http } from "@/api/http";

export interface WizardPlanRequest {
  industry: string;
  style_preference: string;
  features: string[];
  content_types: string[];
  brand_name: string;
  description: string;
  locale: string;
}

export interface WizardPlan {
  recommended_theme: string;
  pages: Array<{
    name: string;
    slug: string;
    description: string;
    title: Record<string, string>;
    layout: string;
    sections: string[];
    sortOrder: number;
  }>;
  color_scheme: {
    primary: string;
    secondary: string;
    background: string;
    text: string;
    rationale?: string;
  };
  suggested_content?: Array<{
    pageSlug: string;
    heading: string;
    subheading: string;
    body: string;
    ctaText: string;
  }>;
  rationale?: string;
  [key: string]: unknown;
}

export interface ColorPalette {
  primary: string;
  secondary: string;
  accent: string;
  background: string;
  [key: string]: string;
}

export interface GeneratedContent {
  title?: string;
  body?: string;
  sections?: unknown[];
  [key: string]: unknown;
}

export async function generateWizardPlan(req: WizardPlanRequest): Promise<WizardPlan> {
  const response = await http.post<{
    recommendedTheme: string;
    pages: Array<{
      slug: string;
      title: Record<string, string>;
      layout: string;
      sections: string[];
      sortOrder: number;
    }>;
    colorScheme: WizardPlan["color_scheme"];
    suggestedContent?: WizardPlan["suggested_content"];
    rationale?: string;
  }>("/admin/wizard/generate-plan", {
    industry: req.industry,
    stylePreference: req.style_preference,
    features: req.features,
    contentTypes: req.content_types,
    brandName: req.brand_name,
    description: req.description,
    locale: req.locale,
  });
  const plan = response.data;
  return {
    recommended_theme: plan.recommendedTheme,
    pages: plan.pages.map((page) => ({
      ...page,
      name: page.title.zh || page.title.en || page.slug,
      description: [page.layout, ...page.sections].filter(Boolean).join(" · "),
    })),
    color_scheme: plan.colorScheme,
    suggested_content: plan.suggestedContent,
    rationale: plan.rationale,
  };
}

export async function applyWizardPlan(plan: WizardPlan): Promise<{ success: boolean; pages_created: number }> {
  const response = await http.post<{ createdPages: string[] }>("/admin/wizard/apply-plan", {
    recommendedTheme: plan.recommended_theme,
    pages: plan.pages.map(({ slug, title, layout, sections, sortOrder }) => ({
      slug,
      title,
      layout,
      sections,
      sortOrder,
    })),
    colorScheme: plan.color_scheme,
    suggestedContent: plan.suggested_content || [],
    rationale: plan.rationale || "",
  });
  return {
    success: true,
    pages_created: response.data.createdPages.length,
  };
}

export async function suggestColors(industry: string, brand_name: string): Promise<ColorPalette> {
  const response = await http.post<ColorPalette>("/admin/wizard/suggest-colors", {
    industry,
    brandName: brand_name,
  });
  return response.data;
}

export async function generateContent(page_type: string, industry: string, locale: string): Promise<GeneratedContent> {
  const response = await http.post<GeneratedContent>("/admin/wizard/generate-content", {
    pageType: page_type,
    industry,
    locale,
  });
  return response.data;
}
